--- a/drw.h	2017-05-24 21:40:16.330373152 -0400
+++ b/drw.h	2017-05-24 21:41:38.549375106 -0400
@@ -50,6 +50,7 @@
 void drw_setscheme(Drw *drw, Scm scm);

 /* Drawing functions */
+void drw_colored_text(Drw *, Scm *, int, int, int, unsigned int, unsigned int, char *);
 void drw_rect(Drw *drw, int x, int y, unsigned int w, unsigned int h, int filled, int invert);
 int drw_text(Drw *drw, int x, int y, unsigned int w, unsigned int h, unsigned int lpad, const char *text, int invert);

--- a/drw.c	2017-05-24 21:40:16.330373152 -0400
+++ b/drw.c	2017-05-24 21:31:01.081359955 -0400
@@ -223,6 +223,33 @@
 }

 void
+drw_colored_text(Drw *drw, Scm *scheme, int numcolors, int x, int y, unsigned int w, unsigned int h, char *text)
+{
+    if (!drw || !drw->scheme)
+		return;
+
+	char *buf = text, *ptr = buf, c = 1, next;
+	int i;
+
+	while (*ptr) {
+		for (i = 0; *ptr < 0 || *ptr > numcolors; i++, ptr++);
+		if (!*ptr)
+			break;
+		c = *ptr;
+		*ptr = ' ';
+        next = *(ptr + 1);
+        *(ptr + 1) = 0;
+		if (i)
+			x = drw_text(drw, x, y, drw_fontset_getwidth(drw, buf), h, 0, buf, 0);
+        *ptr = c;
+        *(ptr + 1) = next;
+		drw_setscheme(drw, scheme[c-1]);
+		buf = ++ptr;
+	}
+	drw_text(drw, x, y, w, h, 0, buf, 0);
+}
+
+void
 drw_rect(Drw *drw, int x, int y, unsigned int w, unsigned int h, int filled, int invert)
 {
 	if (!drw || !drw->scheme)
--- a/dwm.c	2017-05-24 21:41:08.694374396 -0400
+++ b/dwm.c	2017-05-24 21:31:49.071361096 -0400
@@ -722,11 +722,11 @@
 	unsigned int i, occ = 0, urg = 0;
 	Client *c;

-	/* draw status first so it can be overdrawn by tags later */
 	if (m == selmon || 1) { /* status is only drawn on selected monitor */
-		drw_setscheme(drw, scheme[SchemeNorm]);
-		sw = TEXTW(stext) - lrpad + 2; /* 2px right padding */
-		drw_text(drw, m->ww - sw, 0, sw, bh, 0, stext, 0);
+		drw_setscheme(drw, scheme[0]);
+		sw = TEXTW(stext) - lrpad + 4;
+        //sw = drw_get_width(drw, LENGTH(colors), stext) - lrpad + 2; /* 2px right padding */
+		drw_colored_text(drw, scheme, LENGTH(colors), m->ww - sw, 0, sw, bh, stext);
 	}

 	for (c = m->clients; c; c = c->next) {
